packages/block-library/src/file/edit.js

```diff
@@ -171,13 +173,28 @@ function FileEdit( {
 				} }
 			/>
 			<BlockControls>
-				<MediaReplaceFlow
-					mediaId={ id }
-					mediaURL={ href }
-					accept="*"
-					onSelect={ onSelectFile }
-					onError={ onUploadError }
-				/>
+				<ToolbarGroup>
+					<ToolbarItem
+						as={ MediaReplaceFlow }
+						mediaId={ id }
+						mediaURL={ href }
+						accept="*"
+						onSelect={ onSelectFile }
+						onError={ onUploadError }
+					/>
+					<ToolbarItem
+						as={ ClipboardButton }
```

Comment:
Maybe we could try the new `useCopyOnClick` hook here?

Reply:
I tried to use the `useCopyOnClick` hook, but it will trigger an error because it requires using the toolbar button `ref` which is not be rendered yet when the file component loads, then `ref` value will be undefined. 

In case of toolbar which is rendered only on focus `useCopyOnClick` hook requires using a container component which is the exact role of `ClipboardButton`.

Reply:
What if you make it a separate component so `useCopyOnClick` is only called when the button is rendered? `ClipboardButton` just uses `useCopyOnClick` internally, so it should be possible to swap it out.

Reply:
Thank you @ellatrix  for your help. I got your point, i agree it's better to use `useCopyOnClick` hook and reduce code inside 'FileEdit' Component. i've switched to `useCopyOnClick` in daddou-ma/gutenberg@f3d98d7 commit.